src/main/java/org/terasology/Sample/blocks/RomanColumnFamily.java

```diff
@@ -1,38 +1,76 @@
 import org.terasology.world.WorldProvider;
 import org.terasology.world.block.Block;
+import org.terasology.world.block.BlockBuilderHelper;
+import org.terasology.world.block.BlockComponent;
 import org.terasology.world.block.BlockUri;
+import org.terasology.world.block.family.AbstractBlockFamily;
 import org.terasology.world.block.family.BlockFamily;
+import org.terasology.world.block.family.BlockSections;
+import org.terasology.world.block.family.RegisterBlockFamily;
 import org.terasology.world.block.family.UpdatesWithNeighboursFamily;
+import org.terasology.world.block.items.OnBlockItemPlaced;
+import org.terasology.world.block.loader.BlockFamilyDefinition;
+import org.terasology.world.block.shapes.BlockShape;
 
 import java.util.ArrayList;
 import java.util.List;
 
-public class RomanColumnFamily extends UpdatesWithNeighboursFamily {
+@RegisterBlockFamily("romancolumn")
+@BlockSections({"lone_block", "on_top", "on_bottom", "in_a_line"})
+public class RomanColumnFamily extends AbstractBlockFamily implements UpdatesWithNeighboursFamily {
 
-    private TByteObjectMap<Block> blocks;
+    @In
+    WorldProvider worldProvider;
+
+    Iterable<String> CATEGORIES = new ArrayList<String>() {
+
+        private static final long serialVersionUID = 1L;
+
+        {
+
+            add("roman");
+            add("column");
+            add("romancolumn");
+
+        }
+    };
+
+    private TByteObjectMap<Block> BLOCKS;
+
+    BlockUri blockUri;
 
     // Archetype is the base block of the family
     // attachmentSide is the side of the block that the block being placed is going to be attached to
-    public RomanColumnFamily(BlockUri blockUri, List<String> categories, Block archetypeBlock, TByteObjectMap<Block> blocks) {
-        super(null, blockUri, categories, archetypeBlock, blocks, (byte) 63);
-        this.blocks = blocks;
+    public RomanColumnFamily(BlockFamilyDefinition definition, BlockShape shape, BlockBuilderHelper blockBuilder) {
+        super(definition, shape, blockBuilder);
     }
 
-    @Override
-    public Block getBlockForPlacement(WorldProvider worldProvider, BlockEntityRegistry blockEntityRegistry, Vector3i location, Side attachmentSide, Side direction) {
-        return getProperBlock(worldProvider, location);
-    }
+    public RomanColumnFamily(BlockFamilyDefinition definition, BlockBuilderHelper blockBuilder) {
+        super(definition, blockBuilder);
 
-    @Override
-    public Block getBlockForNeighborUpdate(WorldProvider worldProvider, BlockEntityRegistry blockEntityRegistry, Vector3i location, Block oldBlock) {
-        return getProperBlock(worldProvider, location);
+        BLOCKS = new TByteObjectHashMap<Block>();
+
+        blockUri = new BlockUri(definition.getUrn());
+
+        BLOCKS.put((byte) 0, makeABlock(definition, blockBuilder, "lone_block", blockUri, (byte) 0));
+        BLOCKS.put(SideBitFlag.getSide(Side.TOP), makeABlock(definition, blockBuilder, "on_bottom", blockUri, SideBitFlag.getSide(Side.TOP)));
```

Comment:
move these lines to a helper function since you end up repeating getSides for the value in the map and when you set the bitflag.